### PR TITLE
Created aws-py-dynamicresource example

### PR DESCRIPTION
**Pull request by jetvova**

### CODE REVIEW

File: aws-py-dynamicresource/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+pulumi>=2.0.0,<3.0.0
+pulumi-aws>=2.0.0,<3.0.0
```
Comment (justinvp) [inline aws-py-dynamicresource/requirements.txt:2]: We [recently updated](https://github.com/pulumi/examples/pull/761) all the `requirements.txt` in this repo to use version 3.x of the AWS provider.

```suggestion
pulumi-aws>=3.0.0,<4.0.0
```

File: aws-py-dynamicresource/__main__.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+app_gateway = aws.ec2.InternetGateway("app-gateway",
+    vpc_id=app_vpc.id)
+
+app_routetable = aws.ec2.RouteTable("app-routetable",
+    routes=[
+        {
+            "cidr_block": "0.0.0.0/0",
+            "gateway_id": app_gateway.id,
+        }
+    ],
```
Comment (justinvp) [inline aws-py-dynamicresource/__main__.py:38]: `pulumi_aws` 3.2.1 now supports classes for nested data. I've been working on updating all our existing Python examples to use them, but we should also use them for all new Python examples going forward.

```suggestion
    routes=[
        aws.ec2.RouteTableRouteArgs(
            cidr_block="0.0.0.0/0",
            gateway_id=app_gateway.id,
        )
    ],
```

File: aws-py-dynamicresource/__main__.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+		"from_port": 0,
+		"to_port": 65535,
+		"cidr_blocks": ["0.0.0.0/0"],
+    }],
+    egress=[{
+		"protocol": "-1",
+		"from_port": 0,
+		"to_port": 0,
+		"cidr_blocks": ["0.0.0.0/0"],
+    }])
```
Comment (justinvp) [inline aws-py-dynamicresource/__main__.py:62]:
```suggestion
    ingress=[aws.ec2.SecurityGroupIngressArgs(
		protocol="tcp",
		from_port=0,
		to_port=65535,
		cidr_blocks=["0.0.0.0/0"],
    )],
    egress=[aws.ec2.SecurityGroupEgressArgs(
		protocol="-1",
		from_port=0,
		to_port=0,
		cidr_blocks=["0.0.0.0/0"],
    )])
```

**Review by justinvp — APPROVED** [review]

**Comment by cnunciato** [comment]
Merging this to unbreak some links that are pointing to it.

**Pull request merged**